packages/core/src/browser/breadcrumbs/breadcrumbs-renderer.tsx

```diff
@@ -0,0 +1,233 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { injectable, inject, postConstruct } from 'inversify';
+import { ReactRenderer } from '../widgets';
+import { Breadcrumb } from './breadcrumb';
+import { Breadcrumbs } from './breadcrumbs';
+import { BreadcrumbsService } from './breadcrumbs-service';
+import { BreadcrumbRenderer } from './breadcrumb-renderer';
+import PerfectScrollbar from 'perfect-scrollbar';
+import URI from '../../common/uri';
+import { Emitter, Event } from '../../common';
+import { BreadcrumbPopupContainer } from './breadcrumb-popup-container';
+import { DisposableCollection } from '../../common/disposable';
+import { CorePreferences } from '../core-preferences';
+import { Coordinate } from '../context-menu-renderer';
+
+interface Cancelable {
```

Comment:
This needs to be an object so that there can be a shared reference to it on the class and in the method. If it were just a boolean, setting the member to `false` wouldn't affect the variable referred to be the previous method call that needs to be cancelled.